tests/interfaces/records/handlers/records-write.spec.ts

```diff
@@ -813,6 +869,53 @@ describe('RecordsWriteHandler.handle()', () => {
         expect(newWriteReply.status.detail).to.contain('recipient is an immutable property');
       });
 
+      it('should block unauthorized write with author rule', async () => {
+        // scenario: aliceImposter attempts write into Alice's DWN a caption
+        // upon learning the ID of Alice's image
+        // scenario: alice posts an image on the social mebia protocol, then she adds a caption
+        const protocol = 'https://tbd.website/decentralized-web-node/protocols/social-mebia';
+        const protocolDefinition: ProtocolDefinition = socialMebiaProtocolDefinition;
+
+        const alice = await TestDataGenerator.generatePersona();
+        const aliceImposter = await TestDataGenerator.generatePersona();
+
+        const protocolsConfig = await TestDataGenerator.generateProtocolsConfigure({
+          requester: alice,
+          protocol,
+          protocolDefinition
+        });
+
+        // setting up a stub DID resolver
+        TestStubGenerator.stubDidResolver(didResolver, [alice, aliceImposter]);
+
+        const protocolWriteReply = await dwn.processMessage(alice.did, protocolsConfig.message, protocolsConfig.dataStream);
+        expect(protocolWriteReply.status.code).to.equal(202);
+
+        const encodedImage = new TextEncoder().encode('cafe-aesthetic.jpg');
+        const imageRecordsWrite = await TestDataGenerator.generateRecordsWrite({
+          requester : alice,
+          protocol,
+          schema    : socialMebiaProtocolDefinition.labels.image.schema,
+          data      : encodedImage
+        });
+        const imageReply = await dwn.processMessage(alice.did, imageRecordsWrite.message, imageRecordsWrite.dataStream);
+        expect(imageReply.status.code).to.equal(202);
+
+        const encodedCaption = new TextEncoder().encode('bad vibes! >:(');
+        const imageContextId = await imageRecordsWrite.recordsWrite.getEntryId();
+        const captionRecordsWrite = await TestDataGenerator.generateRecordsWrite({
+          requester : aliceImposter,
```

Comment:
Consider merge this the negative test with the positive test above, tagging on the imposter test at the end, as the test is so verbosely similar.

Reply:
Merged